doc/radosgw/multisite.rst

```diff
@@ -0,0 +1,710 @@
+=============
+RGW Multisite
+=============
+
+.. versionadded:: Jewel
+
+From Ceph release Jewel and beyond, you may configure each :term`Ceph Object
+Gateway` to work in an active active zone configuration, allowing for writes to
+non-master zones. Following are the basic terminologies that would be used:
+
+- **Zone**: A zone is *logical* grouping of one or more Ceph Object Gateway
+  instances. There will be one Zone that should be designated as the master zone
+  in a zonegroup, which will handle all bucket and user creation.
+
+- **Zonegroup**: A zonegroup consists of multiple zones, this approximately
+  corresponds to what used to be called as a region in pre Jewel releases for
+  federated deployments. There should be a master zonegroup that will handle
+  changes to the system configuration.
+
+- **Zonegroup map**: A zonegroup map is a configuration structure that holds the
+  map of the entire system, ie. which zonegroup is the master, relationships
+  between different zonegroups and certain configurables like storage policies.
+
+- **Realm**: A realm is a container for zonegroups, this allows for separation
+  of zonegroups themselves between clusters. It is possible to create multiple
+  realms, making it easier to run completely different configurations in the same
+  cluster.
+
+- **Period**: A period holds the configuration structure for the current state
+  of the realm. Every period contains a unique id and an epoch. A period's epoch
+  is incremented on every commit operation. Every realm has an associated
+  current period, holding the current state of configuration of the zonegroups
+  and storage policies. Any configuration change for a non master zone will
+  increment the period's epoch. Changing the master zone to a different zone
+  will trigger the following changes:
+  - A new period is generated with a new period id and epoch of 1
+  - Realm's current period is updated to point to the newly generated period id
+  - Realm's epoch is incremented
+
+Pool Configuration
+==================
+
+About this guide
+================
+
+In this guide we create a single zone group, with three separate zones, which
+actively sync data between them. For the purpose of the guide, we create 2 zones
+in the same cluster, and the third zone in a different cluster. There is no sync
+agent involved for mirroring data changes between the RadosGWs and this allows
+for a much simpler configuration scheme and active-active configurations. Please
+note that metadata operations such as creating a user would still need to go
+through the master zone, however data operations such as creation of buckets
+objects etc. can be handled by any of the zones.
+
+System Keys
+===========
+
+While configuring zones, RadosGW, expects creation of zone a system user, with
+an S3 like access and secret keys. This allows another radosgw instance to pull
```

Comment:
> While configuring zones, RadosGW, expects creation of zone a system user, with an S3 like access and secret keys.

While configuring zones, RadosGW expects creation of a system user, with S3-like access and secret keys.